identity/identity.go

```diff
@@ -544,6 +551,46 @@ func (i *Identity) ValidKeysAtTime(time lamport.Time) []*Key {
 	return result
 }
 
+func (i *Identity) AddKey(key *Key) {
```

Comment:
There is already the `Identity.Mutate()` mechanism to modify an identity. It's a little weird but it allow to update multiple element of an identity in one go (ie, one commit) and update the cache easily. Those key changes should use this mechanism.

Reply:
Done. It's confusing because the "Mutator" name of the class suggests it should be responsible for doing the job, while that is done by a separate function.

Reply:
Yeah that's a good point. A little bit more doc would not hurt here.